lib/ui.rb

```diff
@@ -90,6 +101,20 @@ def warn(s)
       def error(s)
         STDERR.puts s
       end
+
+      private
+
+      def show_progress
+        $stdout.tty? || ENV["SHOW_PROGRESS_OUTPUT"]
```

Comment:
I'll rename the variable.

This is not a variable that should be used by users, though, and it will not be documented. It's merely used for forcing the progress output to be shown in integration tests where there is no tty present. The negated version does not really make sense. Maybe we should call it `FORCE_MACHINERY_PROGRESS_OUTPUT` instead to make that more clear.